starknet/src/server/stream.rs

```diff
@@ -74,6 +79,31 @@ where
 
         ResponseStream::new(data_stream).instrument(stream_span)
     }
+
+    async fn sync_status(self, rpc_url: Url) -> StatusResponse {
```

Comment:
```suggestion
    async fn sync_status(self, rpc_url: Url) -> Result<StatusResponse, tonic::Status> {
```

Let's return a `Result` to avoid unwrapping errors.